server/service/datadeal/validate_data_handler_test.go

```diff
@@ -0,0 +1,246 @@
+
+	dealService.handleValidateData(recorder, req)
+
+	require.Equal(t, http.StatusUnsupportedMediaType, recorder.Result().StatusCode)
+	require.Equal(t, "only application/json is supported\n", recorder.Body.String())
+}
+
+func TestHandlerValidateDataInvalidParameter(t *testing.T) {
+	req := httptest.NewRequest(http.MethodGet, "/v0/data-deal/deals/1/data", bytes.NewReader(defaultData))
+	req = setDefaultURLVars(req)
+
+	req.Header.Set("Content-Type", "application/json")
+
+	recorder := httptest.NewRecorder()
+
+	dealService := makeDataDealService()
+
+	dealService.handleValidateData(recorder, req)
+
+	require.Equal(t, http.StatusBadRequest, recorder.Result().StatusCode)
+	require.Equal(t, "failed to read query parameter\n", recorder.Body.String())
+}
+
+func TestHandlerValidateDataEmptyBody(t *testing.T) {
+	req := httptest.NewRequest(http.MethodGet, "/v0/data-deal/deals/1/data?requester_address="+requesterAddress, nil)
+	req = setDefaultURLVars(req)
+
+	req.Header.Set("Content-Type", "application/json")
+	req.Form = make(map[string][]string)
+	req.Form.Set("requester_address", requesterAddress)
+
+	recorder := httptest.NewRecorder()
+
+	dealService := makeDataDealService()
+
+	dealService.handleValidateData(recorder, req)
+
+	res := recorder.Result()
+	require.Equal(t, http.StatusBadRequest, res.StatusCode)
+	require.Equal(t, "failed to read HTTP request body\n", recorder.Body.String())
+}
+
+func TestHandlerValidateDataInvalidJsonType(t *testing.T) {
+	invalidData := []byte(`{
+		"name": "This is a name",
+		"description": "This is a description, man",
+	}`)
+	req := httptest.NewRequest(http.MethodGet, "/v0/data-deal/deals/1/data?requester_address="+requesterAddress, bytes.NewReader(invalidData))
+	req = setDefaultURLVars(req)
+
+	req.Header.Set("Content-Type", "application/json")
+	req.Form = make(map[string][]string)
+	req.Form.Set("requester_address", requesterAddress)
+
+	recorder := httptest.NewRecorder()
+
+	dealService := makeDataDealService()
+
+	dealService.handleValidateData(recorder, req)
+
+	res := recorder.Result()
+	require.Equal(t, http.StatusForbidden, res.StatusCode)
+	require.Equal(t, "JSON schema validation failed\n", recorder.Body.String())
+}
+
+func TestHandlerValidateDataInvalidBodySchema(t *testing.T) {
+	data := []byte(`{
+		"name": "This is a name",
+		"description": "This is a description, man"
+	}`)
+
+	req := httptest.NewRequest(http.MethodGet, "/v0/data-deal/deals/1/data?requester_address="+requesterAddress, bytes.NewReader(data))
+	req = setDefaultURLVars(req)
+
+	req.Header.Set("Content-Type", "application/json")
+	req.Form = make(map[string][]string)
+	req.Form.Set("requester_address", requesterAddress)
+
+	recorder := httptest.NewRecorder()
+
+	dealService := makeDataDealService()
+
+	dealService.handleValidateData(recorder, req)
+
+	res := recorder.Result()
+
+	require.Equal(t, http.StatusForbidden, res.StatusCode)
+	require.Equal(t, "JSON schema validation failed\n", recorder.Body.String())
+}
```

Comment:
This test seems to be the same with the above test, `TestHandlerValidateDataInvalidJsonType()`